projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,12 +1,56 @@
-import { KeyValue } from '@angular/common';
+import { Dictionary } from '@hypertrace/common';
 import { SelectOption } from '../../select/select-option';
+import { TableFilter } from '../table-api';
 
-export interface SelectFilter {
+export interface SelectControl {
   placeholder?: string;
-  options: SelectOption<KeyValue<string, unknown>>[];
+  default?: SelectOption<TableControlOption>;
+  options: SelectOption<TableControlOption>[];
 }
 
 export interface SelectChange {
-  select: SelectFilter;
-  value: KeyValue<string, unknown>;
+  select: SelectControl;
+  value: TableControlOption;
 }
+
+export interface CheckboxControl {
+  label: string;
+  value: boolean;
```

Comment:
is this the default? If so, does it make sense to match shape with the select control?

Reply:
Yeah, the asymmetry bothers me too. `SelectControl` and `CheckboxControl` are similar, but just distinct enough that they need to be shaped a bit different. It's admittedly confusing. 

The `SelectControl` represents one component dropdown with **1 to n** `options`, each carrying its own `metaValue` for querying based on that selected option... and there can be **0 to n** dropdowns in the table control bar.

The `CheckboxControl` represents a single checkbox and the `options` represent the carried `metaValue` for querying `true` vs `false`. There can also be **1 to n** checkboxes, but they are all clubbed together under a single multiselect dropdown.